eventing/samples/github-events/README.md

```diff
@@ -0,0 +1,162 @@
+    docker build -t {username}/github-events --file=eventing/samples/github-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/github-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in `function.yaml` matches the container you built in the previous
+    step.** Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/function.yaml
+    ```
+
+1.  Check that your service is running using:
+
+    ```shell
+    kubectl get services.serving.knative.dev -o "custom-columns=NAME:.metadata.name,READY:.status.conditions[2].status,REASON:.status.conditions[2].message"
+    NAME              READY     REASON
+    legit             True      <none>
+    ```
+
+1.  Create a [personal access token](https://github.com/settings/tokens) to 
+    GitHub repo that the GitHub source can use to register webhooks with the 
+    GitHub API. Also decide on a token that your code will use to authenticate
+    the incoming webhooks from GitHub (*accessToken*).
+    
+    The token can be named anything you find convenient. This sample requires
+    full `repo` control to be able update the title of the _Pull Request_.
+    The Source requires `admin:repo_hook`, this allows it to create webhooks
+    into repos that your account is allowed to do so. Copy and save this token;
+    GitHub will force you to generate it again if misplaced.
+
+    Here I named my token "EventingSample" and have selected the recommended
+    scopes: 
+
+    ![GitHub UI](personal_access_token.png "GitHub personal access token screenshot")
+
+    Update `eventing/samples/github-events/githubsecret.yaml` with those
+    values. If  your generated access token is `'asdfasfdsaf'` and you choose
+    your *secretToken* as `'personal_access_token_value'`, you'd modify
+    `eventing/samples/github-events/githubsecret.yaml` like so:
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: githubsecret
+    type: Opaque
+    stringData:
+      githubCredentials: >
+        {
+          "accessToken": "asdfasfdsaf",
+          "secretToken": "personal_access_token_value"
+        }
+    ```
+
+    Hint: you can makeup a random *accessToken* with:
+    
+    ```shell
+    head -c 8 /dev/urandom | base64
+    ```
+
+    Then, apply the githubsecret using `kubectl`:
+    
+    ```shell
+    kubectl apply -f eventing/samples/github-events/githubsecret.yaml
+    ```
+
+1.  Create the flow sending GitHub Events to the service:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/flow.yaml
+    ```
+
+1.  Create a PR for the repo you configured the webhook for, and you'll see that
+    the Title will be modified with the suffix `(looks pretty legit)`
+
+
+## Understanding what happened
+
+<!--TODO:
```

Comment:
Can this be fleshed out now? I feel like there's a good start in the introduction, and it would add a lot to the sample.

Reply:
+1 -- much of this should be able to be copied from the `k8sevents` source, with some additional detail on how the Feed for github has a provisioning step which both calls the GitHub API and creates an underlying service (receive adapter). The rest of the text should be able to be shared with the `k8sevent` doc.

Reply:
Requesting this update be a follow-up PR

Reply:
I don't love leaving a blank section in the guide, though. If you're set on not adding anything here for the moment, I'd move the heading inside the TODO so the rendered version of the file doesn't look like it has a blank section. @n3wscott